substrafl/remote/register/register.py

```diff
@@ -413,20 +367,32 @@ def score(self, datasamples, predictions_path, _skip=True):
 
     outputs_metrics = [
         substra.sdk.schemas.FunctionOutputSpec(
-            identifier=OutputIdentifiers.performance,
+            identifier=metric_function_id,
             kind=substra.sdk.schemas.AssetKind.performance,
             multiple=False,
         )
+        for metric_function_id in metric_functions
     ]
 
+    class Metric:
+        def score(self, datasamples, predictions_path, _skip=True):
+            # The _skip argument is needed to match the default signature of methods executed
+            # on substratools_methods.py.
+            return {
+                metric_function_id: metric_functions[metric_function_id](
+                    datasamples=datasamples, predictions_path=predictions_path
+                )
+                for metric_function_id in metric_functions
+            }
+
     remote_struct = RemoteStruct(
         cls=Metric,
         cls_args=[],
         cls_kwargs={},
         remote_cls=RemoteMethod,
         method_name="score",
         method_parameters={},
-        algo_name=metric_name or "metric_" + metric_function.__name__,
+        algo_name="Testing",
```

Comment:
Nitpicking here, but in my mind, testing includes both predicting and evaluating, so I would be more in favour of "Evaluating" or "Computing evaluation metrics"